### DISCUSSION TITLE
Request for Further Guidance on PrNiO3 Charge Order Tutorial

### DISCUSSION THREAD

**Discussion by YUBODRZHANG** [TRIQS/solid_dmft#60] [Q&A]
Dear Professor Hampel,

I am writing to seek further assistance with the second tutorial concerning the charge order in **PrNiO3**. Here, I've encountered a "**rotation.dat**" file that was created by "**_diagonalizing the local d-shell density matrix and identifying the least occupied eigenstates as eg states_**". As a beginner in learning the code, I am finding it challenging to reproduce the data.

Given my current understanding, I am wondering if it would be possible for you to provide additional elaboration or step-by-step instructions that could assist newcomers like myself to better comprehend and follow this process.

While I also have found the Triqs/DFTtools tutorial on **Sr2MgOsO6** to be somewhat illuminating, I still perceive a gap in my understanding that hinders my ability to follow along thoroughly.

I greatly appreciate your advance assistance with this matter, and thank you for your consideration and support.

Best Regards,
Yubo Zhang
Minjiang University

**Comment by the-hampel** [comment]
Dear @YUBODRZHANG,

This is a somewhat technical and tricky question that is very system-dependent. 

**Why and when do you need a rotations.dat file?**
Suppose you have a cubic system with bonds that are somewhat aligned with lattice vectors spanned by 90 degree angles, e.g. SrVO3. Then you never need to supply a rotations file, because the projections to local orbitals that Vasp <chi | KS > (and other DFT codes) will perform are usually done in a Cartesian coordinate frame at the location of the atom of the projection. However, in systems with lower symmetry, such as Pnma or even simpler unit cells but with monoclinic or tetragonal unit cells, the projections are done in a way that they align with the lopes of the orbitals you want to project on. This is not really a problem as long as you are projecting onto a full manifold of a given l quantum number. For example, projecting onto all five d-orbitals should work. However, if you now compute the occupancy matrix of this local projector, it will only be diagonal if your projections are done along the axis aligned with the orbitals. You can now easily tell solid_dmft with `set_rot=hloc_diag` to diagonalize the local Hamiltonian to find the local rotation to the correct basis functions. The problem arises when you have several correlated sites, like in PrNiO3, which should be equivalent to each other but only after some symmetry operations, or when you want to project on a subset of orbitals of an l-shell. Now you lose information and have problems finding a meaningful interaction Hamiltonian. 

**How to find the correct rotations?**
You can either do this geometrically by finding the Euler rotation matrix that rotates in each projection so that it aligns with the orbital axis, or you can simply diagonalize the occupation matrix of all 5 d orbitals first (the one found in the plo converter output) and use the eigenvectors to rotate each correlated site so that the resulting occupation matrix becomes diagonal. Within the wannier90 interface of dft_tools this is done automatically (also in Wien2k and elk I think). However, in Vasp we unfortunately do not have such automatization readily available. 

I hope this helps. Feel free to ask about specific parts, or share your system and we can try to determine the rotations by example.

Best,
Alex

**Reply by YUBODRZHANG** [reply]
Dear Professor Hampel,

I deeply appreciate your thoughtful and insightful response; it has greatly enlightened me on the subject matter. I intend to delve further into the tutorial exercises and will keep you posted once I have focused on a specific system. At this juncture, it seems that Wannier90 might be the most suitable choice for me.

Best,
Yubo
